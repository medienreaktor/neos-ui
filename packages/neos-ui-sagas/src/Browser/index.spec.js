import {testSaga, expectSaga} from 'redux-saga-test-plan';
import {watchContentURIChange, reflectChangeInAddressBar} from './index.js';

describe('watchContentURIChange()', () => {
    it('should be a generator function.', () => {
        expect(typeof watchContentURIChange).toBe('function');
        expect(typeof watchContentURIChange().next).toBe('function');
    });

    it('should not throw an error when running the saga.', () => {
        // ToDo: Currently snapshot testing with takeEvery / takeLatest does not seem to work, investigate.
        expect(() => testSaga(watchContentURIChange).next().next().isDone()).not.toThrow();
    });
});

describe('reflectChangeInAddressBar()', () => {
    let replaceState;

    beforeEach(() => {
        replaceState = jest.spyOn(history, 'replaceState').mockImplementation(jest.fn(() => Promise.resolve()));
    });

    afterEach(() => {
        replaceState.mockRestore();
    });

    it('should be a generator function.', () => {
        expect(typeof reflectChangeInAddressBar).toBe('function');
        expect(typeof reflectChangeInAddressBar().next).toBe('function');
    });

    it('should call the history.replaceState method with the actions payload "contextPath" property.', async () => {
        const result = await expectSaga(reflectChangeInAddressBar, {type: 'DATA', payload: {contextPath: 'fooContextPath'}}).run();

        expect(result.toJSON()).toMatchSnapshot();
    });
});
