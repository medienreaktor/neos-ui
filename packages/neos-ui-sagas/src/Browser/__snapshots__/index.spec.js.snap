// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reflectChangeInAddressBar() should call the history.replaceState method with the actions payload "contextPath" property. 1`] = `
Object {
  "call": Array [
    Object {
      "@@redux-saga/IO": true,
      "CALL": Object {
        "args": Array [
          Object {},
          "",
          "?node=fooContextPath",
        ],
        "context": Object {
          "replaceState": "@@redux-saga-test-plan/json/function/mockConstructor",
        },
        "fn": "@@redux-saga-test-plan/json/function/mockConstructor",
      },
    },
  ],
}
`;
